src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -1038,13 +1040,39 @@ bool ViewProviderSketch::mouseWheelEvent(int delta, const SbVec2s &cursorPos, co
 
     Q_UNUSED(delta);
     Q_UNUSED(cursorPos);
-    Q_UNUSED(viewer);
+
+    if(GridAuto.getValue())
+        updateGridSizeOnZoom(viewer);

```

Comment:
This is very partial as there are much much other ways to change the zoom.
Better integrate in `camSensCB` that is called on camera change (try to minimize number of redraws by finding appropriate criteria).

Reply:
Can you please tell me when this function is called precisely ? Is it everytime the camera moves? ie zoom but also moving sideways?

Reply:
Yes, every time the camera changes. Being it zoom or axis, or even type -- Orthographic/Perspective --. ;)
Why you have to find correct triggers.